### PR TITLE
fix: include interfile dependencies in Snyk Code scans [IDE-182] [IDE-178]

### PR DESCRIPTION
**Pull request by bastiandoetsch**
https://snyksec.atlassian.net/browse/IDE-178

### Description

_Provide description of this PR and changes, if linked Jira ticket doesn't cover it in full._

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

🚨After having merged, please update the CLI go.mod to pull in latest language server.

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: domain/ide/workspace/folder.go
Comment (bastiandoetsch) [inline domain/ide/workspace/folder.go:84]: let's add some comments regarding global/local caching

File: internal/util/identity_test.go
Comment (teodora-sandu) [inline internal/util/identity_test.go:25]:
```suggestion
func Test_GetIssueKey(t *testing.T) {
```

File: infrastructure/code/code.go
Comment (teodora-sandu) [inline infrastructure/code/code.go:212]: I'll need to do a manual test but I noticed in the old code we would run `delete(sc.changedPaths[folderPath], changedPath)` regardless. I wonder if this makes a difference to anything

Comment (teodora-sandu) [reply infrastructure/code/code.go:212]: I don't think it causes any real side effects actually. The only thing I could see happening is if there are multiple scans queued at once and an earlier scan goes through the `changedPaths`, the next scan would still need to see that the left over paths in `changedPaths` are directories and skip them.

File: infrastructure/code/code_test.go
Comment (teodora-sandu) [inline infrastructure/code/code_test.go:908]: Not too sure why this is stored in a variable and the other isn't?

File: infrastructure/code/code_test.go
Comment (teodora-sandu) [inline infrastructure/code/code_test.go:924]: I can see us changing some of the logic in that code_html file or test and then we break this. Would it not be better and clearer to just hardcode it here? It'll also make it more obvious why this test does what it does

**Comment by teodora-sandu** [comment]
I tried to manually test it with a simple repo https://github.com/snyk/target-service and it failed 😕 

```
2024-04-11 17:35:55,018 [  12838]   WARN - #io.snyk.plugin.ui.SnykBalloonNotifications - scanning error for project target-service, emptying cache.Data: SnykScanParams(status=error, product=code, folderPath=/Users/teodorasandu/Documents/repos/target-service, issues=null)
```

Screenshot 2024-04-11 at 17 34 41

Not sure if I've done something wrong or I've found an edge case by accident.

**Review by teodora-sandu — COMMENTED** [review]
I was just about to approve it when the manual test I ran errored. I'm not sure if it's my setup or not. I won't block it if you want to get another opinion, since I'm off tomorrow 🙇

File: infrastructure/code/code_test.go
Comment (bastiandoetsch) [reply infrastructure/code/code_test.go:924]: Yeah, I was lazy ;)

**Comment by bastiandoetsch** [comment]
I'll try it out tomorrow, didn't see the issue though. Maybe something related to the new flow - we'll see.

**Comment by bastiandoetsch** [comment]
> I tried to manually test it with a simple repo https://github.com/snyk/target-service and it failed 😕
> 
> ```
> 2024-04-11 17:35:55,018 [  12838]   WARN - #io.snyk.plugin.ui.SnykBalloonNotifications - scanning error for project target-service, emptying cache.Data: SnykScanParams(status=error, product=code, folderPath=/Users/teodorasandu/Documents/repos/target-service, issues=null)
> ```
> 
> Screenshot 2024-04-11 at 17 34 41
> 
> Not sure if I've done something wrong or I've found an edge case by accident.
> 
> This is my setup: Screenshot 2024-04-11 at 17 38 41

I think it is caused by the workspace API not ready yet in prod. 
```
2024-04-12 08:32:54,065 [ 486578]   INFO - Snyk Language Server - 2024-04-12T08:32:54+02:00 INF - creating workspace bundleHash=xxx path=/Users/bdoetsch/workspace/juice-shop repositoryUri=https://github.com/juice-shop/juice-shop.git

2024-04-12 08:32:54,238 [ 486751]   INFO - Snyk Language Server - 2024-04-12T08:32:54+02:00 ERR processResults - Product returned an error error= product="Snyk Code"
```

File: domain/snyk/scanner.go
Comment (bastiandoetsch) [reply domain/snyk/scanner.go:272]: 0-values in golang are tricky to compare, and if we don't pass a pointer, we cannot compare to nil. So I'm using the ID field as substitute for checking if the issue is empty. We could use title or key, too - it's not really important which of the mandatory fields are checked.

File: infrastructure/code/code.go
Comment (bastiandoetsch) [reply infrastructure/code/code.go:212]: Yeah, it should not make a difference. Also, it's cleaner to have the guard first in the loop, I think.

**Review by teodora-sandu — APPROVED** [review]

**Pull request merged**
